lldb/test/API/functionalities/breakpoint/swift_exception/TestExpressionErrorBreakpoint.py

```diff
@@ -26,81 +26,73 @@ class TestSwiftErrorBreakpoint(TestBase):
 
     @decorators.skipIfLinux  # <rdar://problem/30909618>
     @swiftTest
-    def test_swift_error_no_pattern(self):
+    def test_swift_error_no_typename(self):
         """Tests that swift error throws are correctly caught by the Swift Error breakpoint"""
-        self.build()
-        self.do_test(lldb.SBStringList(), True)
+        self.do_tests(None, True)
 
     @swiftTest
-    def test_swift_error_matching_base_pattern(self):
+    def test_swift_error_matching_base_typename(self):
         """Tests that swift error throws are correctly caught by the Swift Error breakpoint"""
-        self.build()
-        pattern = lldb.SBStringList()
-        pattern.AppendString("exception-typename")
-        pattern.AppendString("EnumError")
-        self.do_test(pattern, True)
+        self.do_tests("EnumError", True)
 
     @swiftTest
-    def test_swift_error_matching_full_pattern(self):
+    def test_swift_error_matching_full_typename(self):
         """Tests that swift error throws are correctly caught by the Swift Error breakpoint"""
-        self.build()
-        pattern = lldb.SBStringList()
-        pattern.AppendString("exception-typename")
-        pattern.AppendString("a.EnumError")
-        self.do_test(pattern, True)
+        self.do_tests("a.EnumError", True)
 
     @swiftTest
-    def test_swift_error_bogus_pattern(self):
+    def test_swift_error_bogus_typename(self):
         """Tests that swift error throws are correctly caught by the Swift Error breakpoint"""
-        self.build()
-        pattern = lldb.SBStringList()
-        pattern.AppendString("exception-typename")
-        pattern.AppendString("NoSuchErrorHere")
-        self.do_test(pattern, False)
+        self.do_tests("NoSuchErrorHere", False)
 
     def setUp(self):
         TestBase.setUp(self)
-        self.main_source = "main.swift"
-        self.main_source_spec = lldb.SBFileSpec(self.main_source)
-
-    def do_test(self, patterns, should_stop):
-        """Tests that swift error throws are correctly caught by the Swift Error breakpoint"""
+        self.build()
 
+    def do_tests(self, typename, should_stop):
+        self.do_test(typename, should_stop, self.create_breakpoint_with_api)
+        self.do_test(typename, should_stop, self.create_breakpoint_with_command)
+
+    def create_breakpoint_with_api(self, target, typename):
+        types = lldb.SBStringList()
+        if typename:
+            types.AppendString("exception-typename")
+            types.AppendString(typename)
+        return target.BreakpointCreateForException(
+            lldb.eLanguageTypeSwift, False, True, types)
+
+    def create_breakpoint_with_command(self, target, typename):
+        cmd = "breakpoint set -E swift"
+        if typename:
+            cmd += " --exception-typename " + typename
+        self.runCmd(cmd)
```

Comment:
Might be better to make a `lldbutil.run_break_for_exception` along the lines of "run_break_set_by_name, etc.  Then you can use the check_breakpoint_result to pull the breakpoint out and make sure that the setting actually worked.

Reply:
Should we be adding those on the Swift branch? (no strong opinion either way)

Reply:
Break for exception works for C++ & ObjC, so there's no reason not to.  The Object type would be an extra argument, so you don't need to bring along any explicit swiftness.

Reply:
I have added [`lldbutil.run_break_set_by_exception`](https://github.com/apple/llvm-project/pull/4027/commits/a368c060ee0f972065904879233bedb99747f6cd). I will open a diff to upstream llvm too.